hepcrawl/parsers/arxiv.py

```diff
@@ -0,0 +1,361 @@
+        dois = [
+            {'doi': value, 'material': 'publication'} for value in doi_values
+        ]
+
+        return dois
+
+    @property
+    def licenses(self):
+        licenses = self.root.xpath('.//license/text()').extract()
+        return [{'url':license, 'material': self.material} for license in licenses]
+
+    @property
+    def material(self):
+        return 'preprint'
+
+    @property
+    def number_of_pages(self):
+        comments = '; '.join(self.root.xpath('.//comments/text()').extract())
+        pages, _, __ = _parse_comments_info(comments)
+
+        return maybe_int(pages)
+
+    @property
+    def publication_info(self):
+        publication_info = {
+            'material': 'publication',
+            'pubinfo_freetext': self.pubinfo_freetext,
+        }
+
+        return publication_info
+
+    @property
+    def pubinfo_freetext(self):
+        return self.root.xpath('.//journal-ref/text()').extract_first()
+
+    @property
+    def title(self):
+        return self.root.xpath('.//title/text()').extract_first()
+
+    @staticmethod
+    def get_root_node(arxiv_record):
+        """Get a selector on the root ``article`` node of the record.
+
+        This can be overridden in case some preprocessing needs to be done on
+        the XML.
+
+        Args:
+            arxiv_record(Union[str, scrapy.selector.Selector]): the record in arXiv format.
+
+        Returns:
+            scrapy.selector.Selector: a selector on the root ``<article>``
+                node.
+        """
+        if isinstance(arxiv_record, six.string_types):
+            root = get_node(arxiv_record)
+        else:
+            root = arxiv_record
+        root.remove_namespaces()
+
+        return root
+
+    @property
+    def public_note(self):
+        comments = '; '.join(self.root.xpath('.//comments/text()').extract())
+        _, notes, __ = _parse_comments_info(comments)
+
+        return notes
+
+    @property
+    def private_note(self):
+        _, __, warning = _get_authors_and_collaborations(self.root)
+
+        return warning
+
+    @property
+    def report_numbers(self):
+        report_numbers = self.root.xpath('.//report-no/text()').extract()
+        rns = []
+        for rn in report_numbers:
+            rns.extend(rn.split(', '))
+
+        return rns
+    
+    @property
+    def arxiv_eprint(self):
+        return self.root.xpath('.//id/text()').extract_first()
+
+    @property
+    def arxiv_categories(self):
+        categories = self.root.xpath('.//categories/text()').extract_first()
+
+        if categories:
+            return categories.split()
+
+        return []
+
+    @property
+    def document_type(self):
+        comments = '; '.join(self.root.xpath('.//comments/text()').extract())
+        _, __, doctype = _parse_comments_info(comments)
+
+        return doctype
+
+    @property
+    def source(self):
+        return 'arXiv'
+
+
+def _parse_comments_info(comments):
```

Comment:
I would turn this into a `staticmethod`, as it's some internal thing used by the parser.